benchmark/bipedal-timings.cpp

```diff
@@ -0,0 +1,349 @@
+                                                                               terminalCostModel);
+
+  boost::shared_ptr<crocoddyl::ActionModelAbstract> runningModel =
+      boost::make_shared<crocoddyl::IntegratedActionModelEuler>(runningDAM, 1e-3);
+  boost::shared_ptr<crocoddyl::ActionModelAbstract> terminalModel =
+      boost::make_shared<crocoddyl::IntegratedActionModelEuler>(terminalDAM, 1e-3);
+
+  std::vector<boost::shared_ptr<crocoddyl::ActionModelAbstract> > runningModels(N, runningModel);
+
+  boost::shared_ptr<crocoddyl::ShootingProblem> problem =
+      boost::make_shared<crocoddyl::ShootingProblem>(x0, runningModels, terminalModel);
+  std::vector<Eigen::VectorXd> xs(N + 1, x0);
+
+  /***************************************************************/
+
+  boost::shared_ptr<crocoddyl::ActionDataAbstract> runningModel_data = runningModel->createData();
+  boost::shared_ptr<crocoddyl::DifferentialActionDataAbstract> runningDAM_data = runningDAM->createData();
+  crocoddyl::DifferentialActionDataContactFwdDynamics* d =
+      static_cast<crocoddyl::DifferentialActionDataContactFwdDynamics*>(runningDAM_data.get());
+  boost::shared_ptr<crocoddyl::ActuationDataAbstract> actuation_data = actuation->createData();
+  boost::shared_ptr<crocoddyl::CostDataAbstract> goalTrackingCost_data = goalTrackingCost->createData(&d->multibody);
+  boost::shared_ptr<crocoddyl::CostDataAbstract> xRegCost_data = xRegCost->createData(&d->multibody);
+  boost::shared_ptr<crocoddyl::CostDataAbstract> uRegCost_data = uRegCost->createData(&d->multibody);
+
+  boost::shared_ptr<crocoddyl::CostDataSum> runningCostModel_data = runningCostModel->createData(&d->multibody);
+
+  boost::shared_ptr<crocoddyl::ActivationModelAbstract> activationQuad = xRegCost->get_activation();
+  boost::shared_ptr<crocoddyl::ActivationDataAbstract> activationQuad_data = activationQuad->createData();
+
+  /********************************************************************/
+
+  double duration = 0;
+  PINOCCHIO_ALIGNED_STD_VECTOR(Eigen::VectorXd) x1s;  // (T, state->rand());
+  PINOCCHIO_ALIGNED_STD_VECTOR(Eigen::VectorXd) x2s;  // (T, state->rand());
+  PINOCCHIO_ALIGNED_STD_VECTOR(Eigen::VectorXd) us;   // (T, state->rand());
+  PINOCCHIO_ALIGNED_STD_VECTOR(Eigen::VectorXd) dxs(T, Eigen::VectorXd::Zero(2 * model.nv));
+
+  for (size_t i = 0; i < T; ++i) {
+    x1s.push_back(state->rand());
+    x2s.push_back(state->rand());
+    us.push_back(Eigen::VectorXd(actuation->get_nu()));
+  }
+
+  /*********************State**********************************/
+
+  duration = 0;
+  timer.reset();
+  SMOOTH(T) {
+    pinocchio::difference(model, x1s[_smooth].head(model.nq), x2s[_smooth].head(model.nq),
+                          dxs[_smooth].head(model.nv));
+  }
+  duration = timer.get_us_duration();
+  std::cout << "pinocchio::difference (in us):\t\t\t" << duration / T << " us" << std::endl;
```

Comment:
I think here Rohan is just trying to find the most consuming part of the current implementation. It shoud be removed when things will be done.
Again, for code generation, we cannot expect to do a correct implementation in a first row. It will be necessary to iterate over it, as done in Pinocchio. Let's go ahead and try the code.

Reply:
Yes. This code belongs in pinocchio. But since we are using pinocchio stuff inside crocoddyl, it is good to know how much of an additional computational burder is being added by crocoddyl and how much is being added by pinocchio. 
I prefer to keep this here.